lib/net.js

```diff
@@ -1514,6 +1522,15 @@ function onconnection(err, clientHandle) {
     return;
   }
 
+  // Check to see if the remote address has been blocked.
+  // TODO(@jasnell): Performance would be better if this was checked down
+  // in the C++ layer as close to libuv as possible, but that's a bit of
+  // a larger change. Will do that as a follow up step.
+  const addr = {};
+  err = clientHandle.getpeername(addr);
+  if (err || self[kBlockList].check(addr.address, addr.family.toLowerCase()))
+    return clientHandle.close();
```

Comment:
Note: There's more of a performance penalty doing the check here in this way. Ideally, the check would be done at the C++ level before the callback is invoked but that's a bit of a larger change. I'm going to benchmark this approach first, then work on moving the check down into the C++ layer.